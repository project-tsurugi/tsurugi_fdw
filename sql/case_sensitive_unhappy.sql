/* Test case: unhappy path - Case sensitivity */
-- Test setup: DDL of the Tsurugi
SELECT tg_execute_ddl('
  CREATE TABLE fdw_case_table (
    col INT,
    Col INT
  )
', 'tsurugidb');

-- Test setup: DDL of the PostgreSQL
CREATE FOREIGN TABLE fdw_case_table (
  col integer,
  "Col" integer
) SERVER tsurugidb;

-- Test
INSERT INTO fdw_Case_Table (col) VALUES (1000);
INSERT INTO fdw_Case_table (col) VALUES (2000);
INSERT INTO fdw_CASE_table (col) VALUES (3000);
INSERT INTO fdw_CASE_TABLE (col) VALUES (4000);
INSERT INTO public.fdw_Case_Table (col) VALUES (5000);
INSERT INTO "Public".fdw_case_table (col) VALUES (6000);

INSERT INTO fdw_case_table (COL) VALUES (7000);
INSERT INTO fdw_case_table ("COL") VALUES (8000);
INSERT INTO fdw_case_table (col, "Col") VALUES (9000, 9);

UPDATE fdw_Case_Table SET col = 1000;
UPDATE fdw_Case_table SET col = 2000;
UPDATE fdw_CASE_table SET col = 3000;
UPDATE fdw_CASE_TABLE SET col = 4000;
UPDATE public.fdw_Case_Table SET col = 5000;
UPDATE "Public".fdw_case_table SET col = 6000;
UPDATE fdw_case_table SET COL = 100;
UPDATE fdw_case_table SET col = 100 WHERE COL > 5000;
UPDATE fdw_case_table SET col = 100 WHERE "COL" > 5000;

DELETE FROM fdw_Case_Table;
DELETE FROM fdw_Case_table;
DELETE FROM fdw_CASE_table;
DELETE FROM fdw_CASE_TABLE;
DELETE FROM public.fdw_Case_Table;
DELETE FROM "Public".fdw_case_table;
DELETE FROM fdw_case_table WHERE COL > 5000;
DELETE FROM fdw_case_table WHERE "COL" > 5000;

-- Test teardown: DDL of the PostgreSQL
DROP FOREIGN TABLE fdw_case_table;
-- Test teardown: DDL of the Tsurugi
SELECT tg_execute_ddl('DROP TABLE fdw_case_table', 'tsurugidb');

