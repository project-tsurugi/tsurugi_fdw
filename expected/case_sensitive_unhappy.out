/* Test case: unhappy path - Case sensitivity */
-- Test setup: DDL of the Tsurugi
SELECT tg_execute_ddl('
  CREATE TABLE fdw_case_table (
    col INT,
    Col INT
  )
', 'tsurugidb');
 tg_execute_ddl 
----------------
 CREATE TABLE
(1 row)

-- Test setup: DDL of the PostgreSQL
CREATE FOREIGN TABLE fdw_case_table (
  col integer,
  "Col" integer
) SERVER tsurugidb;
-- Test
INSERT INTO fdw_Case_Table (col) VALUES (1000);
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_Case_Table" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: INSERT INTO fdw_Case_Table (col) VALUES (1000)
INSERT INTO fdw_Case_table (col) VALUES (2000);
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_Case_table" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: INSERT INTO fdw_Case_table (col) VALUES (2000)
INSERT INTO fdw_CASE_table (col) VALUES (3000);
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_CASE_table" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: INSERT INTO fdw_CASE_table (col) VALUES (3000)
INSERT INTO fdw_CASE_TABLE (col) VALUES (4000);
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_CASE_TABLE" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: INSERT INTO fdw_CASE_TABLE (col) VALUES (4000)
INSERT INTO public.fdw_Case_Table (col) VALUES (5000);
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_Case_Table" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: INSERT INTO fdw_Case_Table (col) VALUES (5000)
INSERT INTO "Public".fdw_case_table (col) VALUES (6000);
ERROR:  relation "Public.fdw_case_table" does not exist
LINE 1: INSERT INTO "Public".fdw_case_table (col) VALUES (6000);
                    ^
INSERT INTO fdw_case_table (COL) VALUES (7000);
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:column_not_found message:"column is not found: fdw_case_table.COL" location:<input>:1:29+3)
CONTEXT:  SQL query: INSERT INTO fdw_case_table (COL) VALUES (7000)
INSERT INTO fdw_case_table ("COL") VALUES (8000);
ERROR:  column "COL" of relation "fdw_case_table" does not exist
LINE 1: INSERT INTO fdw_case_table ("COL") VALUES (8000);
                                    ^
INSERT INTO fdw_case_table (col, "Col") VALUES (9000, 9);
UPDATE fdw_Case_Table SET col = 1000;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_Case_Table" is not found" location:<input>:1:8+14)
CONTEXT:  SQL query: UPDATE fdw_Case_Table SET col = 1000
UPDATE fdw_Case_table SET col = 2000;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_Case_table" is not found" location:<input>:1:8+14)
CONTEXT:  SQL query: UPDATE fdw_Case_table SET col = 2000
UPDATE fdw_CASE_table SET col = 3000;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_CASE_table" is not found" location:<input>:1:8+14)
CONTEXT:  SQL query: UPDATE fdw_CASE_table SET col = 3000
UPDATE fdw_CASE_TABLE SET col = 4000;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_CASE_TABLE" is not found" location:<input>:1:8+14)
CONTEXT:  SQL query: UPDATE fdw_CASE_TABLE SET col = 4000
UPDATE public.fdw_Case_Table SET col = 5000;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_Case_Table" is not found" location:<input>:1:8+14)
CONTEXT:  SQL query: UPDATE fdw_Case_Table SET col = 5000
UPDATE "Public".fdw_case_table SET col = 6000;
ERROR:  relation "Public.fdw_case_table" does not exist
LINE 1: UPDATE "Public".fdw_case_table SET col = 6000;
               ^
UPDATE fdw_case_table SET COL = 100;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:column_not_found message:"UPDATE SET target column is not found: fdw_case_table.COL" location:<input>:1:27+3)
CONTEXT:  SQL query: UPDATE fdw_case_table SET COL = 100
UPDATE fdw_case_table SET col = 100 WHERE COL > 5000;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:variable_not_found message:"variable "COL" is not found, searching for {column_variable, field_variable, schema_variable}" location:<input>:1:43+3)
CONTEXT:  SQL query: UPDATE fdw_case_table SET col = 100 WHERE COL > 5000
UPDATE fdw_case_table SET col = 100 WHERE "COL" > 5000;
ERROR:  column "COL" does not exist
LINE 1: UPDATE fdw_case_table SET col = 100 WHERE "COL" > 5000;
                                                  ^
DELETE FROM fdw_Case_Table;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_Case_Table" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: DELETE FROM fdw_Case_Table
DELETE FROM fdw_Case_table;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_Case_table" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: DELETE FROM fdw_Case_table
DELETE FROM fdw_CASE_table;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_CASE_table" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: DELETE FROM fdw_CASE_table
DELETE FROM fdw_CASE_TABLE;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_CASE_TABLE" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: DELETE FROM fdw_CASE_TABLE
DELETE FROM public.fdw_Case_Table;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:table_not_found message:"table "fdw_Case_Table" is not found" location:<input>:1:13+14)
CONTEXT:  SQL query: DELETE FROM fdw_Case_Table
DELETE FROM "Public".fdw_case_table;
ERROR:  relation "Public.fdw_case_table" does not exist
LINE 1: DELETE FROM "Public".fdw_case_table;
                    ^
DELETE FROM fdw_case_table WHERE COL > 5000;
ERROR:  Failed to execute remote SQL.
HINT:  Failed to execute the statement on Tsurugi. error: SERVER_ERROR(13)
Tsurugi Error: SYMBOL_ANALYZE_EXCEPTION (SQL-03004: compile failed with error:variable_not_found message:"variable "COL" is not found, searching for {column_variable, field_variable, schema_variable}" location:<input>:1:34+3)
CONTEXT:  SQL query: DELETE FROM fdw_case_table WHERE COL > 5000
DELETE FROM fdw_case_table WHERE "COL" > 5000;
ERROR:  column "COL" does not exist
LINE 1: DELETE FROM fdw_case_table WHERE "COL" > 5000;
                                         ^
-- Test teardown: DDL of the PostgreSQL
DROP FOREIGN TABLE fdw_case_table;
-- Test teardown: DDL of the Tsurugi
SELECT tg_execute_ddl('DROP TABLE fdw_case_table', 'tsurugidb');
 tg_execute_ddl 
----------------
 DROP TABLE
(1 row)

